src/main/java/ru/testtask/controller/UserController.java

```diff
@@ -1,39 +1,115 @@
 package ru.testtask.controller;
 
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
-import org.springframework.security.core.context.SecurityContextHolder;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 import ru.testtask.dto.UserDTO;
+import ru.testtask.dto.ViewUserDTO;
+import ru.testtask.exception.NameAlreadyExistsException;
 import ru.testtask.model.User;
 import ru.testtask.service.UserService;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 
 @RestController
+@RequestMapping("api/users")
 public class UserController {
 
     @Autowired
     private UserService userService;
 
+    @Autowired
+    private ModelMapper modelMapper;
 
     @PreAuthorize("hasAuthority('ADMIN')")
```

Comment:
Aggreed to use config instead `@PreAuthorize("hasAuthority('ADMIN')")` for all methods